src/components/AccountDrawer/UniwalletModal.tsx

```diff
@@ -39,44 +38,32 @@ const Divider = styled.div`
 `
 
 export default function UniwalletModal() {
-  const open = useModalIsOpen(ApplicationModal.UNIWALLET_CONNECT)
-  const toggle = useToggleUniwalletModal()
-
+  const { pending, cancelActivation } = useActivateConnection()
   const [uri, setUri] = useState<string>()
+  const open = pending.connection?.type === ConnectionType.UNIWALLET && !isIOS && !!uri
```

Comment:
Question: why do we have `!isIOS` here and `!!uri`? Is `open` a noop in other cases? Maybe it's better to handle this outside of the component. Otherwise, seeing this code is a bit confusing as to how the code behaves when `open` is false (despite `pending.connection?.type === ConnectionType.UNIWALLET` being true).

Reply:
Good point, I'll move the isIOS check to the parent & add a comment explaining the check for URI here